pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BitmapBasedFilterOperator.java

```diff
@@ -86,11 +86,26 @@ public BaseFilterBlock nextFilterBlock(BlockId BlockId) {
       default:
         throw new UnsupportedOperationException("Regex is not supported");
     }
-    ImmutableRoaringBitmap[] bitmaps = new ImmutableRoaringBitmap[dictionaryIds.length];
-    for (int i = 0; i < dictionaryIds.length; i++) {
-      bitmaps[i] = invertedIndex.getImmutable(dictionaryIds[i]);
+
+    // For realtime use case, it is possible that inverted index has not yet generated for the given dict id, so we
+    // filter out null bitmaps
+    int length = dictionaryIds.length;
+    List<ImmutableRoaringBitmap> bitmaps = new ArrayList<>(length);
+    for (int dictionaryId : dictionaryIds) {
+      ImmutableRoaringBitmap bitmap = invertedIndex.getImmutable(dictionaryId);
+      if (bitmap != null) {
+        bitmaps.add(bitmap);
+      }
     }
-    bitmapBlock = new BitmapBlock(dataSource.getOperatorName(), dataSourceBlock.getMetadata(), startDocId, endDocId, bitmaps, exclusion);
+
+    // Log size diff to verify the fix
+    int numBitmaps = bitmaps.size();
+    if (numBitmaps != length) {
+      LOGGER.info("Not all inverted indexes are generated, numDictIds: {}, numBitmaps: {}", length, numBitmaps);
```

Comment:
This log may appear for every query to every realtime consuming segment, is that right? Can we change this to debug? Do we want to add a metric if this is important?